translated: docs/developer-docs/integrations/bitcoin

hokosugi opened the pull request:
## 疑問点・不明点・相談したい点

技術的な点で不明なところがあって、うまく翻訳出来ったところがあります。

* bitcoin-how-it-works.md
    - 14行目 `non-trivial` を「自明性がない」としましたが、数学表現でしっくり来ません（index.mdの12行目も同じ単語）
    - 20行目 `outgoing transaction` を「発信用トランザクション」としましたが、イマイチです。P2P層に送るinput txの集合体のことだと思うのですが、なんと訳せばよいものか？
    - 34行目 ` set of recent blocks` を「最近のブロックのセット」としましたが、出来れば「直近」にしたほうが伝わるのではないかと思いましたが、原文通りではないので悩んでいます。
    - 36行目 以下の訳が難しかったです。
        - but UTXOs therein must be efficiently considered for calculating the UTXOs of a given address in addition to those in the UTXO set.
         - （訳）しかし、UTXO セット内のものに加えて、与えられたアドレスの UTXO を計算するために、そこにある UTXO を効率的に考慮しなければなりません。
         - ちょっと、意味が通じていないと思います。フォーク等の対策のためにブロックを維持しないといけないけど、効率化も必要と言いたいと思うですが、ややこしくなってしまっています。
     - 48行目 「署名サブネット」と「ECDSAサブネット」は同じサブネットと思うのですが、違うのですかね？（勝手な質問です）
     - 63行目 ここの訳が難しいです。
         - In other words, if the number of confirmations is c, an output is returned if it occurred in a transaction with at least c confirmations and there is no transaction that spends the same output with at least c confirmations.
         - （訳）言い換えれば、コンファーム回数が c の場合、少なくとも c 回のコンファームがあるトランザクションは出力され、少なくとも c 回のコンファームを経た同じ出力のトランザクションはありません。
         - これは全てのc回コンファームされたTXが出力されている、残りはない！といいたいと思うのですが、よくわかりません。
     - 68行目 パーセンタイルのことがまだ良くわかってなくて間違えているかもしれません。確認お願いします。
     - 72行目 上手いタイトルになっていません。
         - Development, Pre-Production, and Production Environment
         - （訳）開発・プレプロダクション・プロダクション環境
 * local-development.md
      - 186行目 `coinbase maturity` をそのままにしましたが、どうでしょうか？実は何のことだか分かっていません。
     
以上です。長いですが、宜しくおねがいします。
指摘をしてくだされば、こちらで修正してコミットいたします！！

hokosugi committed:
translated: docs/developer-docs/integrations/bitcoin

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+Canister 自身は、新しい閾値 ECDSA プロトコルを用いて ECDSA 鍵を持つことができるため、Bitcoin を受け取り、保持できます。また、Canister はBitcoin のトランザクションを作成し、Bitcoin API を介して Bitcoin ネットワークに送信することができます。Canister は閾値 ECDSA の機能を利用して、Bitcoin ネットワークに送信する取引で閾値 ECDSA の署名をリクエストして UTXO を消費することができます。閾値 ECDSA は、IC の Chain Key 暗号のツールボックスのプロトコルを拡張したものです。IC の閾値 ECDSA プロトコルの詳細は、閾値 ECDSA のドキュメントページ [ここ](../t-ecdsa/t-ecdsa-how-it-works.md)で見ることができます。
+
+プロトコルレベルの Bitcoin 統合と閾値 ECDSA プロトコルは、それぞれマネージメント Canister 上で API を公開しています。それらの API はエンジニアが IC 上で Bitcoin のスマートコントラクトを記述するために使用するシステムレベルの API です。 Bitcoin の UTXO とトランザクションの概念を中心に設計された低レベルの API であり、その使用は自明性はなく、Bitcoin の仕組みについて深い理解が必要とされます。
```
確かに非自明は数学的表現なので、「簡単でない」ことがわかる表現であれば良いと思います。
```suggestion
プロトコルレベルの Bitcoin 統合と閾値 ECDSA プロトコルは、それぞれマネージメント Canister 上で API を公開しています。それらの API はエンジニアが IC 上で Bitcoin のスマートコントラクトを記述するために使用するシステムレベルの API です。 Bitcoin の UTXO とトランザクションの概念を中心に設計された低レベルの API であり、その使用は簡単ではなく、Bitcoin の仕組みについて深い理解が必要とされます。
```

hoosan replied:
```suggestion
プロトコルレベルの Bitcoin 統合と閾値 ECDSA プロトコルは、それぞれマネージメント Canister 上で API を公開しています。それらの API はエンジニアが IC 上で Bitcoin のスマートコントラクトを記述するために使用するシステムレベルの API です。 Bitcoin の UTXO とトランザクションの概念を中心に設計された低レベルの API であり、その使用は簡単なものではなく、Bitcoin の仕組みについて深い理解が必要とされます。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+# Bitcoin 統合の仕組み ー 技術的背景
+
+Bitcoin を IC 内で統合するには、2つの高度な技術的課題を解決する必要がありました：
+（1）IC と Bitcoin ネットワークのプロトコルレベルでの統合、（2）新しい閾値 ECDSA プロトコルの開発。
```
日本語としては一つの文の方が自然と思いましたがいかがでしょうか
```suggestion
Bitcoin を IC 内で統合するには、（1）IC と Bitcoin ネットワークのプロトコルレベルでの統合、（2）新しい閾値 ECDSA プロトコルの開発という、2つの高度な技術的課題を解決する必要がありました。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+**革新的の ECDSA プロトコル**
+
+Canister 自身は、新しい閾値 ECDSA プロトコルを用いて ECDSA 鍵を持つことができるため、Bitcoin を受け取り、保持できます。また、Canister はBitcoin のトランザクションを作成し、Bitcoin API を介して Bitcoin ネットワークに送信することができます。Canister は閾値 ECDSA の機能を利用して、Bitcoin ネットワークに送信する取引で閾値 ECDSA の署名をリクエストして UTXO を消費することができます。閾値 ECDSA は、IC の Chain Key 暗号のツールボックスのプロトコルを拡張したものです。IC の閾値 ECDSA プロトコルの詳細は、閾値 ECDSA のドキュメントページ [ここ](../t-ecdsa/t-ecdsa-how-it-works.md)で見ることができます。
```
文章に自然に含める形に修正を行いました
```suggestion
Canister 自身は、新しい閾値 ECDSA プロトコルを用いて ECDSA 鍵を持つことができるため、Bitcoin を受け取り、保持できます。また、Canister はBitcoin のトランザクションを作成し、Bitcoin API を介して Bitcoin ネットワークに送信することができます。Canister は閾値 ECDSA の機能を利用して、Bitcoin ネットワークに送信する取引で閾値 ECDSA の署名をリクエストして UTXO を消費することができます。閾値 ECDSA は、IC の Chain Key 暗号のツールボックスのプロトコルを拡張したものです。IC の閾値 ECDSA プロトコルの詳細は、[こちらの](../t-ecdsa/t-ecdsa-how-it-works.md) 閾値 ECDSA のドキュメントページで見ることができます。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+プロトコルレベルの Bitcoin 統合と閾値 ECDSA プロトコルは、それぞれマネージメント Canister 上で API を公開しています。それらの API はエンジニアが IC 上で Bitcoin のスマートコントラクトを記述するために使用するシステムレベルの API です。 Bitcoin の UTXO とトランザクションの概念を中心に設計された低レベルの API であり、その使用は自明性はなく、Bitcoin の仕組みについて深い理解が必要とされます。
+
+次に、Bitcoin との直接統合の背後にある上記の技術の高レベルな概要を説明します。詳細は、[Bitcoin page on the Internet Computer Wiki](https://wiki.internetcomputer.org/wiki/Bitcoin_integration) および [threshold ECDSA documentation page](../t-ecdsa/t-ecdsa-how-it-works.md) を参照してください。
```
高レベルが２つの意味に取れてしまう文なので、「大まかな」ではどうでしょうか
```suggestion
次に、Bitcoin との直接統合の背後にある上記の技術の大まかな概要を説明します。詳細は、[Bitcoin page on the Internet Computer Wiki](https://wiki.internetcomputer.org/wiki/Bitcoin_integration) および [threshold ECDSA documentation page](../t-ecdsa/t-ecdsa-how-it-works.md) を参照してください。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+## プロトコルレベルでの IC と Bitcoin ネットワークとの統合
+
+Internet Computer プロトコルと Bitcoin プロトコルを統合し、2つのネットワーク間の直接的な技術統合が可能になりました。この統合は、任意の数の Internet Computer サブネットでアクティブにすることができます。当初は、1つの専用の Bitcoin-activated サブネットのみが存在し、任意のサブネット上の Canister からの Bitcoin API へのリクエストは、この単一の Bitcoin-activated サブネットにルーティングされることになります。この統合は、2つの重要な目的を果たすものです：
```
「当初」という言葉からは過去の事項を示すニュアンスを感じるので、未来の（there will be）事項に関しては「最初」の方が適していると思いました
```suggestion
Internet Computer プロトコルと Bitcoin プロトコルを統合し、2つのネットワーク間の直接的な技術統合が可能になりました。この統合は、任意の数の Internet Computer サブネットでアクティブにすることができます。まず最初は 1つの専用の Bitcoin-activated サブネットのみが存在することになり、任意のサブネット上の Canister からの Bitcoin API へのリクエストは、この単一の Bitcoin-activated サブネットにルーティングされることになります。この統合は、2つの重要な目的を果たすものです：
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+## プロトコルレベルでの IC と Bitcoin ネットワークとの統合
+
+Internet Computer プロトコルと Bitcoin プロトコルを統合し、2つのネットワーク間の直接的な技術統合が可能になりました。この統合は、任意の数の Internet Computer サブネットでアクティブにすることができます。当初は、1つの専用の Bitcoin-activated サブネットのみが存在し、任意のサブネット上の Canister からの Bitcoin API へのリクエストは、この単一の Bitcoin-activated サブネットにルーティングされることになります。この統合は、2つの重要な目的を果たすものです：
+-    Bitcoin の UTXO セットを取得し、Internet Computer のレプリカされたステートでチェーン上に保持し、Canister が発行した Bitcoin アカウントが行う UTXO セットと残高の問い合わせにレスポンスができるようにすること。
```
「レプリカされた」という言葉が気になりました（レプリカはIC用語ではソフトウェアを示す認識です）
```suggestion
-    Bitcoin の UTXO セットを取得し、Internet Computer の複製されたステートに含めてチェーン上に保持し、Canister が発行した Bitcoin アカウントが行う UTXO セットと残高の問い合わせにレスポンスができるようにすること。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+Internet Computer プロトコルと Bitcoin プロトコルを統合し、2つのネットワーク間の直接的な技術統合が可能になりました。この統合は、任意の数の Internet Computer サブネットでアクティブにすることができます。当初は、1つの専用の Bitcoin-activated サブネットのみが存在し、任意のサブネット上の Canister からの Bitcoin API へのリクエストは、この単一の Bitcoin-activated サブネットにルーティングされることになります。この統合は、2つの重要な目的を果たすものです：
+-    Bitcoin の UTXO セットを取得し、Internet Computer のレプリカされたステートでチェーン上に保持し、Canister が発行した Bitcoin アカウントが行う UTXO セットと残高の問い合わせにレスポンスができるようにすること。
+-   Canister の署名付き Bitcoin 取引を受け入れ、 Bitcoin ネットワークに送信すること。
```
トランザクションで統一するのが良いのではないでしょうか（他の箇所もです）
```suggestion
-   Canister の署名付き Bitcoin トランザクションを受け入れ、 Bitcoin ネットワークに送信すること。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+**コンポーネント**
+
+ Bitcoin が有効なサブネットでは、マネージメント Canister の API を介して、マネージメント Canister の一部として、すなわちレプリカの一部として実装された *BTC Canister*（Bitcoin Canister）が、Canister にアクセスできるようにされています。BTC Canister は、オンチェーンの Bitcoin 関連のステート、すなわち UTXO セット、フォーク・ソリューションを可能にする最新の Bitcoin ブロック、および発信用トランザクションを保持します。
```
マスタリング・ビットコインではfork resolutionは「フォークの解決」と訳しているので、ここでも解決とするのが良いと思います（同時に複数のマイナーがブロックを採掘したとき、最新のブロックで拡張することでフォークを解決（確定）させる話のことです）。
発信用という部分ですが、確かにあまり対応する用語がないので、わかりやすさを重視して「外に出ていく」と訳すのはどうでしょうか。
```suggestion
 Bitcoin が有効なサブネットでは、マネージメント Canister の API を介して、マネージメント Canister の一部として、すなわちレプリカの一部として実装された *BTC Canister*（Bitcoin Canister）が、Canister にアクセスできるようにされています。BTC Canister は、オンチェーンの Bitcoin 関連のステート、すなわち UTXO セット、フォークの解決を可能にする最新の Bitcoin ブロック、および外に出ていくトランザクションを保持します。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+** Bitcoin UTXO セットの維持管理**
+
+BTC Canister とアダプターは統合されており、IC のプロトコルスタックを介して相互に通信します。BTC Canister は、受け取った最新の Bitcoin ブロックの後継ブロックを Bitcoin アダプターに要求します。サブネットの各レプリカのアダプターは、要求されたブロックを Bitcoin ネットワークから取得し、ブロック作成レプリカのアダプターは、要求されたブロックをコンセンサスを経て Bitcoin Canister に提供します。 Bitcoin Canister は、プルーフ・オブ・ワークを検証し、ブロックのトランザクションを抽出し、トランザクションから UTXO を抽出し、レプリケートステートに維持されている UTXO セットを更新するという形式でブロックを処理して、消費された UTXO とトランザクションによって作成された UTXO らを反映させます。U ...
```
replicated なので、カタカナにするのであればレプリケーテッドが正しいと思いますが、「複製された」でも良いかと思いました。
最近の、に関しては私は違和感ありません。
```suggestion
BTC Canister とアダプターは統合されており、IC のプロトコルスタックを介して相互に通信します。BTC Canister は、受け取った最新の Bitcoin ブロックの後継ブロックを Bitcoin アダプターに要求します。サブネットの各レプリカのアダプターは、要求されたブロックを Bitcoin ネットワークから取得し、ブロック作成レプリカのアダプターは、要求されたブロックをコンセンサスを経て Bitcoin Canister に提供します。 Bitcoin Canister は、プルーフ・オブ・ワークを検証し、ブロックのトランザクションを抽出し、トランザクションから UTXO を抽出し、複製されたステートに維持されている UTXO セットを更新するという形式でブロックを処理して、消費された UTXO とトランザクションによって作成された UTXO らを反映させます。UTXO セットと、まだ UTXO セットに吸収されていない最近のブロックのセットは、UTXO と残高に対するリクエストに応答するために使用されます。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+BTC Canister とアダプターは統合されており、IC のプロトコルスタックを介して相互に通信します。BTC Canister は、受け取った最新の Bitcoin ブロックの後継ブロックを Bitcoin アダプターに要求します。サブネットの各レプリカのアダプターは、要求されたブロックを Bitcoin ネットワークから取得し、ブロック作成レプリカのアダプターは、要求されたブロックをコンセンサスを経て Bitcoin Canister に提供します。 Bitcoin Canister は、プルーフ・オブ・ワークを検証し、ブロックのトランザクションを抽出し、トランザクションから UTXO を抽出し、レプリケートステートに維持されている UTXO セットを更新するという形式でブロックを処理して、消費された UTXO とトランザクションによって作成された UTXO らを反映させます。U ...
+
+実装には、フォークを安全に解決し、様々な種類の攻撃から保護する重要な複雑な領域があります。例えば、フォークを解決できるようにするために、Bitcoin Canister が維持している UTXO セットには、まだ吸収されていない一定数の UTXO セットがあり、そのために Bitcoin ブロックを維持する必要があります。しかし、UTXO セット内のものに加えて、与えられたアドレスの UTXO を計算するために、そこにある UTXO を効率的に考慮しなければなりません。
```
少し読みにくさを感じたので修正しました。

確かに維持しているブロックの中には"まだ吸収されていない UTXO セット"があるので文章としては正しいと思うのですが、フォークを解決するために維持する必要があるとここで言っているのはブロックです。（"まだ吸収されていない一定数の UTXO セット"は、原文には記載されていないと思います）

「そこにある」でも通じると思いますが、ここで言っている「そこ」とは「Bitcoin blocks which are not yet absorbed」のことなので、読みやすさのために置き換えました。
```suggestion
フォークを安全に解決したり様々な種類の攻撃から保護するために、セキュリティの領域において非常に複雑な実装になっています。例えば、フォークを解決できるようにするために、Bitcoin Canister は UTXO セットにまだ吸収されていない Bitcoin ブロックを一定数維持する必要があります。しかし、与えられたアドレスの UTXO を計算するためには、UTXO セット内のものに加えて、吸収させずに維持しているブロック内にある UTXO を効率的に考慮しなければなりません。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+閾値 ECDSA とは、閾値暗号を用いた ECDSA 署名プロトコルの実装のことです。閾値 ECDSA プロトコルでは、ECDSA 秘密鍵は複数の当事者間で秘密共有され、当事者のうち適格な定足数のみがそれぞれの秘密鍵シェアを使って署名を生成することができます。秘密鍵は復元された形では決して存在せず、秘密共有された形でのみ存在します。鍵の生成により、各当事者の秘密鍵シェアが生成されます。
+
+Internet Computer の Chain Key 暗号ツールボックスの一部として実装された閾値 ECDSA プロトコルは、単一のマスター秘密鍵を使用し、そこから BIP-32 ライクな鍵派生を使用して Canister の鍵を派生させることができます。各 Canister は、Canister ID を使用して導き出された1つのルート鍵を持っており、BIP-32 の下位互換性のある拡張を使用して任意の数の追加の Canister 鍵を導き出すことができます。
```
backwardは後方と訳されます
```suggestion
Internet Computer の Chain Key 暗号ツールボックスの一部として実装された閾値 ECDSA プロトコルは、単一のマスター秘密鍵を使用し、そこから BIP-32 ライクな鍵派生を使用して Canister の鍵を派生させることができます。各 Canister は、Canister ID を使用して導き出された1つのルート鍵を持っており、BIP-32 の後方互換性のある拡張を使用して任意の数の追加の Canister 鍵を導き出すことができます。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+Internet Computer の Chain Key 暗号ツールボックスの一部として実装された閾値 ECDSA プロトコルは、単一のマスター秘密鍵を使用し、そこから BIP-32 ライクな鍵派生を使用して Canister の鍵を派生させることができます。各 Canister は、Canister ID を使用して導き出された1つのルート鍵を持っており、BIP-32 の下位互換性のある拡張を使用して任意の数の追加の Canister 鍵を導き出すことができます。
+
+Internet Computer 上の閾値 ECDSA が最初にデプロイされるのは、Canister の署名要求に答える1つの署名サブネットです。署名サブネットでは鍵を管理する Canister のみがこの鍵での署名を要求できます。すべての API コールは、ECDSA サブネットに到達するために Xnet トラフィックを通過する必要があり、それに応じて余分なレイテンシーが発生します。
```
「署名サブネット」と「ECDSAサブネット」は同じという理解で良いと思います

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+## API
+
+Bitcoin 統合により、以下のマネージメント用 Canister API が利用可能になります（閾値 ECDSA APIについては、[そのドキュメントページ](../t-ecdsa/t-ecdsa-how-it-works.md)と[インターフェース仕様](../../../references/ic-interface-spec.md)で説明しています）。Bitcoin 関連の各メソッドは、Bitcoin `mainnet` と `testnet` のどちらを使用するかを指定する必要があります。
```
```suggestion
Bitcoin 統合により、以下のマネージメント Canister の API が利用可能になります（閾値 ECDSA APIについては、[そのドキュメントページ](../t-ecdsa/t-ecdsa-how-it-works.md)と[インターフェース仕様](../../../references/ic-interface-spec.md)で説明しています）。Bitcoin 関連の各メソッドは、Bitcoin `mainnet` と `testnet` のどちらを使用するかを指定する必要があります。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+Bitcoin 統合により、以下のマネージメント用 Canister API が利用可能になります（閾値 ECDSA APIについては、[そのドキュメントページ](../t-ecdsa/t-ecdsa-how-it-works.md)と[インターフェース仕様](../../../references/ic-interface-spec.md)で説明しています）。Bitcoin 関連の各メソッドは、Bitcoin `mainnet` と `testnet` のどちらを使用するかを指定する必要があります。
+
+-   `bitcoin_get_utxos`: Bitcoin アドレスと Bitcoin ネットワーク（メインネットまたはテストネット）を指定する必要がある `get_utxos_request` を与えると、Bitcoin コンポーネントで有効になっている Bitcoin ブロックチェーンの現在のビューに基づいて、この関数は指定した Bitcoin ネットワーク内の指定したアドレスに関するすべての unspent transaction outputs（UTXOs）を返します。UTXO はブロックの高さの降順でソートされて返されます。<br/>
```
「未使用トランザクションアウトプット」はマスタリング・ビットコインでもそのように訳されているので修正しました。
```suggestion
-   `bitcoin_get_utxos`:  `get_utxos_request` が与えられると（Bitcoin アドレスと Bitcoin ネットワーク（メインネットまたはテストネット）を指定する必要があります）、Bitcoin コンポーネントで有効になっている Bitcoin ブロックチェーンの現在のビューに基づいて、この関数は指定した Bitcoin ネットワーク内の指定したアドレスに関するすべての未使用トランザクションアウトプット（UTXOs）を返します。UTXO はブロックの高さの降順でソートされて返されます。<br/>
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+Bitcoin 統合により、以下のマネージメント用 Canister API が利用可能になります（閾値 ECDSA APIについては、[そのドキュメントページ](../t-ecdsa/t-ecdsa-how-it-works.md)と[インターフェース仕様](../../../references/ic-interface-spec.md)で説明しています）。Bitcoin 関連の各メソッドは、Bitcoin `mainnet` と `testnet` のどちらを使用するかを指定する必要があります。
+
+-   `bitcoin_get_utxos`: Bitcoin アドレスと Bitcoin ネットワーク（メインネットまたはテストネット）を指定する必要がある `get_utxos_request` を与えると、Bitcoin コンポーネントで有効になっている Bitcoin ブロックチェーンの現在のビューに基づいて、この関数は指定した Bitcoin ネットワーク内の指定したアドレスに関するすべての unspent transaction outputs（UTXOs）を返します。UTXO はブロックの高さの降順でソートされて返されます。<br/>
+オプションのフィルターパラメータを使用して、返される UTXO のセットを制限することができます。パラメータに最小限のコンファーム数を与えるか、または多くの UTXO を持つアドレスに対してページ分割が使用されている場合は、ページの参照を与えます。最初のケースでは、少なくとも与えられたコンファーム回数を持つ UTXO のみが返されます。つまり、コンファーム回数がこの回数より少ないトランザクションは考慮されません。言い換えれば、コンファーム回数が c の場合、少なくとも c 回のコンファームがあるトランザクションは出力され、少なくとも c 回のコンファームを経た同じ出力のトランザクションはありません。<br/>
```
修正しましたのでご確認ください。
意味が理解しづらい場合には他の言い回し考えてみます🙏
```suggestion
オプションのフィルターパラメータを使用して、返される UTXO のセットを制限することができます。パラメータに最小の承認数を指定するか、または多くの UTXO を持つアドレスに対してページ分割されている場合はページへの参照を指定することができます。前者のケースでは、指定された数以上の承認数を持つ UTXO のみが返されます。つまり、承認数がこの回数より少ないトランザクションは考慮されません。言い換えれば、承認数が c の場合、少なくとも c 回承認されているトランザクションによって生じたアウトプットのうち、消費されてないものが返されます。ただし、返されたアウトプットは承認回数が c 回未満のトランザクションにはすでに消費されている可能性があります。<br/>
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+オプションのフィルターパラメータを使用して、返される UTXO のセットを制限することができます。パラメータに最小限のコンファーム数を与えるか、または多くの UTXO を持つアドレスに対してページ分割が使用されている場合は、ページの参照を与えます。最初のケースでは、少なくとも与えられたコンファーム回数を持つ UTXO のみが返されます。つまり、コンファーム回数がこの回数より少ないトランザクションは考慮されません。言い換えれば、コンファーム回数が c の場合、少なくとも c 回のコンファームがあるトランザクションは出力され、少なくとも c 回のコンファームを経た同じ出力のトランザクションはありません。<br/>
+オプションのフィルターなしの `get_utxos_request` は、ブロックチェーン全体を考慮したリクエストになり、これは `min_confirmations` を 0 に設定することと同じです。
+-  `bitcoin_get_balance`: Bitcoin アドレスと Bitcoin ネットワーク (メインネットまたはテストネット) を指定する `get_balance_request` を受け取ると、この関数は、指定した Bitcoin ネットワークにおけるこのアドレスの現在の残高を Satoshi (10<sup>8</sup> Satoshi = 1 Bitcoin) で返します。`bitcoin_get_utxos` と同じアドレス形式がサポートされています。
+-  `bitcoin_send_transaction` は Bitcoin 取引の Blob と Bitcoin ネットワーク（メインネットまたはテストネット）を指定しなければなりません。`send_transaction_request` が与えられると、いくつかのチェックが行われ、成功すればトランザクションは Bitcoin ネットワークに転送されます。
```
```suggestion
-  `bitcoin_send_transaction`: `send_transaction_request` を与えると（Bitcoin トランザクション の Blob と Bitcoin ネットワーク（メインネットまたはテストネット）を指定する必要があります）、いくつかのチェックが行われ、成功すればトランザクションは Bitcoin ネットワークに転送されます。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+オプションのフィルターなしの `get_utxos_request` は、ブロックチェーン全体を考慮したリクエストになり、これは `min_confirmations` を 0 に設定することと同じです。
+-  `bitcoin_get_balance`: Bitcoin アドレスと Bitcoin ネットワーク (メインネットまたはテストネット) を指定する `get_balance_request` を受け取ると、この関数は、指定した Bitcoin ネットワークにおけるこのアドレスの現在の残高を Satoshi (10<sup>8</sup> Satoshi = 1 Bitcoin) で返します。`bitcoin_get_utxos` と同じアドレス形式がサポートされています。
+-  `bitcoin_send_transaction` は Bitcoin 取引の Blob と Bitcoin ネットワーク（メインネットまたはテストネット）を指定しなければなりません。`send_transaction_request` が与えられると、いくつかのチェックが行われ、成功すればトランザクションは Bitcoin ネットワークに転送されます。
+-  `bitcoin_get_current_fee_percentiles`：Bitcoin ネットワークの取引手数料は、保留中の取引数に基づいて動的に変化します。Bitcoin 取引を作成する際に、Canister が適切な手数料を決定することが可能でなければなりません。<br/> この関数は、100の手数料パーセンタイルをミリ秒単位で返します。
```
この文は次の行の一部です
```suggestion
-  `bitcoin_get_current_fee_percentiles`：Bitcoin ネットワークの取引手数料は、保留中の取引数に基づいて動的に変化します。Bitcoin 取引を作成する際に、Canister が適切な手数料を決定することが可能でなければなりません。<br/>
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+-  `bitcoin_get_balance`: Bitcoin アドレスと Bitcoin ネットワーク (メインネットまたはテストネット) を指定する `get_balance_request` を受け取ると、この関数は、指定した Bitcoin ネットワークにおけるこのアドレスの現在の残高を Satoshi (10<sup>8</sup> Satoshi = 1 Bitcoin) で返します。`bitcoin_get_utxos` と同じアドレス形式がサポートされています。
+-  `bitcoin_send_transaction` は Bitcoin 取引の Blob と Bitcoin ネットワーク（メインネットまたはテストネット）を指定しなければなりません。`send_transaction_request` が与えられると、いくつかのチェックが行われ、成功すればトランザクションは Bitcoin ネットワークに転送されます。
+-  `bitcoin_get_current_fee_percentiles`：Bitcoin ネットワークの取引手数料は、保留中の取引数に基づいて動的に変化します。Bitcoin 取引を作成する際に、Canister が適切な手数料を決定することが可能でなければなりません。<br/> この関数は、100の手数料パーセンタイルをミリ秒単位で返します。
+この関数は、過去10,000件の取引、つまり過去約4～10ブロックの取引について、millisatoshi/byte (10<sup>3</sup> millisatoshi = 1 satoshi) で測定した100件の手数料パーセンタイルを返します。これは通常、支払われるべき手数料の確かな指標となりますが、手数料のパーセンタイルの計算では Bitcoin mempool を考慮しないことに注意してください。
```
パーセンタイルはデータを順位で100等分したものです。
今回で言うと10,000件分のトランザクションの手数料を順位ソートして
「安い方から100件（1%分）みていくと100件目の手数料はいくら、200件（2%分）みていくといくら・・」と100%分まで手数料を計算してテーブルを作ります。
すると、「（安い方から数えて）90%のトランザクションが使用したfeeは○○だから、そのくらいfeeとして与えておけば9割がた大丈夫だろう」みたいな判断ができます。

![image](https://user-images.githubusercontent.com/40290137/196867974-193926c3-c3b7-4519-9902-036a1f649adc.png)


```suggestion
この関数は、過去10,000件の取引、つまり過去約4～10ブロックの取引について、millisatoshi/byte (10<sup>3</sup> millisatoshi = 1 satoshi) で測定した手数料のパーセンタイルを返します。これは通常、支払われるべき手数料の確かな指標となりますが、手数料のパーセンタイルの計算では Bitcoin mempool を考慮しないことに注意してください。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+ Bitcoin 連携 API の詳細については、[インタフェイス仕様](../../../references/ic-interface-spec.md) を参照してください。
+
+## 開発・プレプロダクション・プロダクション環境
```
日本語ではそれぞれに環境をつけたほうがわかりやすいかもしれません。
プレプロダクション環境という言葉はあまり聞かないように思いますが（検証環境とかステージング環境の方がよく聞く）、意味は通じるので良いと思います。
```suggestion
## 開発環境・プレプロダクション環境・プロダクション環境
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+### ローカル SDK
+
+一般的な Canister の開発ワークフローでは、IC 上の Canister は、開発時に Motoko または Rust の Canister SDK を使ってローカル環境でコンパイル・実行されます。このように、SDK は開発ワークフローの最初の段階、つまり環境となります。この SDK は Bitcoin 統合 Canister API と閾値 ECDSA マネージメント Canister API の両方に対応できるようになっています。
```
or はここでは「あるいは」とするのが良いと思います
```suggestion
一般的な Canister の開発ワークフローでは、IC 上の Canister は、開発時に Motoko または Rust の Canister SDK を使ってローカル環境でコンパイル・実行されます。このように、SDK は開発ワークフローの最初のステージ、あるいは環境となります。この SDK は Bitcoin 統合 Canister API と閾値 ECDSA マネージメント Canister API の両方に対応できるようになっています。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+一般的な Canister の開発ワークフローでは、IC 上の Canister は、開発時に Motoko または Rust の Canister SDK を使ってローカル環境でコンパイル・実行されます。このように、SDK は開発ワークフローの最初の段階、つまり環境となります。この SDK は Bitcoin 統合 Canister API と閾値 ECDSA マネージメント Canister API の両方に対応できるようになっています。
+
+それぞれ Bitcoin テストネットと Bitcoin メインネットと統合する機能の IC デプロイとは対照的に、SDK はリグレッションテスト（regtest）モードでローカルに動作する bitcoind ノードと統合されています。regtest モードで bitcoind を使用することは、Bitcoin の開発において好ましい方法です。開発者をできるだけ容易にするために、私たちは SDK を regtest モードの bitcoind と統合し、IC に最高の Bitcoin 開発体験をもたらしました。スマートコントラクトの開発と自動テストの両方が、この設定でまずローカル環境で行われます。
```
「開発を容易にする」あるいは「開発者を手助けする」のような表現が良いと思います
```suggestion
それぞれ Bitcoin テストネットと Bitcoin メインネットと統合する機能の IC デプロイとは対照的に、SDK はリグレッションテスト（regtest）モードでローカルに動作する bitcoind ノードと統合されています。regtest モードで bitcoind を使用することは、Bitcoin の開発において好ましい方法です。開発者をできる限り手助けするために、私たちは SDK を regtest モードの bitcoind と統合し、IC に最高の Bitcoin 開発体験をもたらしました。このセットアップにより、スマートコントラクトの開発と自動テストの両方が、まずローカル環境で行われます。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/bitcoin-how-it-works.md:
```diff
@@ -1,3 +1,106 @@
+
+### IC テストネット上の Bitcoin
+
+スマートコントラクトが受け入れテストの準備ができると、IC 上にデプロイされます（パブリックな IC テストネットがないことを思い出してください）。Bitcoin テストネットに接続するために Bitcoin API セットを使用します。これはローカル開発で使用するのと同じ設定です。この設定は、Bitcoin testnet を使用して Dapp の受け入れテストを実行するために使用され、リアル価値は危険にさらされていません。
```
```suggestion
スマートコントラクトが受け入れテストの準備ができると、IC 上にデプロイされます（パブリックな IC テストネットがないことを思い出してください）。Bitcoin テストネットに接続するために Bitcoin API セットを使用します。これはローカル開発で使用するのと同じ設定です。この設定では Dapp の受け入れテストを実行するために Bitcoin テストネットを使用します。つまり、実際に価値のあるものは危険にさらされません。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/index.md:
```diff
@@ -1,3 +1,28 @@
+
+-   *Bitcoin のスマートコントラクト：*  Canister は Bitcoin ネットワーク上の Bitcoin を直接保持できるため、エンジニアは Canister を使用して強力な Bitcoin のスマートコントラクトを実装することが可能になります。どのような Canister のスマートコントラクトでも、Bitcoin のスマートコントラクト機能を提供できるようになりました。例えば、ユーザーが秘密鍵を管理する必要のない生体認証によるオンチェーン Bitcoin ウォレットや、ソーシャル Dapps を使用してユーザーがピアツーピアで Bitcoin 取引を行うことができる socialFi などです。
+-   Internet Computer 上の分散型取引所での *Bitcoin の取引*。
+-   SNS を活用した DAO が IC 上のサービスを分散化する際に、Bitcoin を使用して *分散化セール* でトークンを購入すること。
```
```suggestion
-   SNS を活用した DAO が IC 上のサービスを分散化する際に、Bitcoin を使用して *分散型セール (Decentralization Sale)* でトークンを購入すること。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/index.md:
```diff
@@ -1,3 +1,28 @@
+-   SNS を活用した DAO が IC 上のサービスを分散化する際に、Bitcoin を使用して *分散化セール* でトークンを購入すること。
+-   *Chain Key Bitcoin （ckBTC）*は Wrapped Bitcoin の発展型で、Bitcoin メインネットの Bitcoin 機能のリリースに伴い、IC 上で利用可能になります。ckBTC は IC 上で最も簡単に Bitcoin を扱うことができ、IC 上の Bitcoin に興味を持つ多くの人にとって正しい選択になるかもしれません。なお、ckBTC は今後数カ月間の Bitcoin の一般提供（ GA ）リリースと、IC 上での Bitcoin メインネットのローンチによってのみ利用可能となる予定です。
+
+これらは Bitcoin との連携機能のほんの一例に過ぎません。この機能によって開かれる無限の可能性は、あなたの想像力以外に限界はありません。このドキュメントでは、Bitcoin を使用した独自の Dapp を実装するための機能の使用方法について説明します。また、Bitcoin 統合 API は、UTXO と Bitcoin トランザクションのレベルで動作する低レベルの API であり、使用することに自明性がないことに注意してください。
```
直訳な感じがしましたので少し言い換えました
```suggestion
これらは Bitcoin との連携機能のほんの一例に過ぎません。あなたの想像力が及ぶ限り、無限の可能性が開かれています。このドキュメントでは、Bitcoin を使用した独自の Dapp を実装するための機能の使用方法について説明します。また、Bitcoin 統合 API は、UTXO と Bitcoin トランザクションのレベルで動作する低レベルの API であり、使用は簡単ではないことに注意してください。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/local-development.md:
```diff
@@ -1,3 +1,276 @@
+
+## Bitcoin Dapp をデプロイする
+
+これでローカルな Bitcoin ネ ットワークが構築できたので Bitcoin Dapps の開発とローカルでのテストを始める準備ができています。
```
```suggestion
これでローカルな Bitcoin ネットワークが構築できたので Bitcoin Dapps の開発とローカルでのテストを始める準備ができています。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/local-development.md:
```diff
@@ -1,3 +1,276 @@
+これは、1ブロックのマイニングで得た報酬です。
+
+:::note
+マイニングした BTC はお客様のローカル Bitcoin ネットワーク内でのみ有効であり、他の場所で使用したり、使ったりすることはできません。
```
```suggestion
マイニングした BTC はあなたのローカル Bitcoin ネットワーク内でのみ有効であり、他の場所で使用したり、使ったりすることはできません。
```

hoosan commented on docs/developer-docs/integrations/bitcoin/local-development.md:
```diff
@@ -1,3 +1,276 @@
+
+### ステートをリセットする
+
+ローカルにある Bitcoin のステートをすべて削除して、一からやり直すと便利なことが多いです。ゼロから始めることができます。そのためには：
```
```suggestion
ローカルにある Bitcoin のステートをすべて削除して、一からやり直すとよいことがしばしばあります。そのためには：
```

hoosan commented on docs/developer-docs/integrations/bitcoin/local-development.md:
```diff
@@ -1,3 +1,276 @@
+マイニングした BTC はお客様のローカル Bitcoin ネットワーク内でのみ有効であり、他の場所で使用したり、使ったりすることはできません。
+:::
+
+### coinbase maturity
```
Coinbase maturityというのは、マイニング報酬で得たアウトプット（Coinbase output）が使用できるようになるまでに100ブロックの追加マイニングが必要というルールです。
https://docs.grin.mw/wiki/miscellaneous/coinbase-maturity-rule/
あえて訳すと「Coinbase 成熟度」とでも言えると思いますが、訳している文章は見つからなかったので、そのままで良いと思います。

hoosan reviewed (changes requested):
翻訳ありがとうございます。かなり技術的にディープなところだったと思います。疑問点は適宜コメントをご参照ください🙏

hokosugi committed:
Apply suggestions from code review

レビューありがとうございます。分かりやすかったです。
マスタリングビットコインを参照すべきでしたね、ツメが甘いと再認識しました。次に活かしたいと思います。ひとつコミットしていないのがあります。イマイチ、ピンと来ないのでもう一度確認してコメント入れたいと思います。

Co-authored-by: hoosan <40290137+hoosan@users.noreply.github.com>

hoosan commented:
承認数のところどうでしょう？意味としては、c回以上承認されたトランザクションから発生したUTXOを取得するというものです。その際、もしかすると現時点でUTXOはすでに使われているかもしれませんが、最低c回承認されたトランザクションには使われてないことは保証されてます。（c-1回以下承認されたトランザクションにはすでに消費されてる可能性があります）

hokosugi commented:
なるほど、少し分かりました。
それでも私には少し理解が難しいので、以下のような意味合いで提案します。文意が合っていればよいのですが、間違えていた場合は、そのままコミットしていただいて結構です。こちらは勝手に文意を掴み取りに参ります！

言い換えれば、承認数が c の場合、少なくとも c 回承認されているトランザクションによって生じたアウトプットのうち、消費されてないアウトプット（c 回未満承認されているトランザクションは除外）が返されます。

hoosan commented:
文意良さそうです！文章としては以下はどうでしょう？

言い換えれば、承認数が c の場合、少なくとも c 回承認されているトランザクションによって生じたアウトプットのうち、消費されてないものが返されます。ただし、返されたアウトプットは承認回数がc回未満のトランザクションにはすでに消費されている可能性があります。

hokosugi commented:
はい！
少し曖昧かもしれませんが、原文も同様なので良いかと思います。アダプターから上手く拾えないUTXOがあって開発チームが試行錯誤をしていたので、また文章が更新されるかもしれません。追々、分かって来ると思います。
それでお願いします。

hoosan committed:
Apply suggestions from code review
